ecl/hql/hqlattr.cpp

```diff
@@ -3776,6 +3776,71 @@ ITypeInfo * setStreamedAttr(ITypeInfo * _type, bool setValue)
     }
 }
 
+//---------------------------------------------------------------------------------
+double queryLikelihood(IHqlExpression * expr)
+{
+    IInterface * match = meta.queryExistingProperty(expr, EPlikelihood);
+    if (match)
+    {
+        IHqlExpression * likelihoodExpr = static_cast<IHqlExpression *>(match);
+        IValue * value = likelihoodExpr->queryChild(0)->queryValue();
```

Comment:
I'm not sure the queryChild(0) is correct.  Try calling the function multiple times and check the same results is returned